util/llpcShaderModuleHelper.cpp

```diff
@@ -28,20 +28,327 @@
 * @brief LLPC source file: Implementation of LLPC utility class ShaderModuleHelper
 ***********************************************************************************************************************
 */
+
+#include "SPIRVFunction.h"
+#include "SPIRVInstruction.h"
+#include "SPIRVModule.h"
+#include "SPIRVType.h"
+
 #include "llpcShaderModuleHelper.h"
 #include "llpcDebug.h"
 #include "llpcUtil.h"
+#include "llpcInternal.h"
 #include "spirvExt.h"
 #include <unordered_set>
 #include <set>
 
 #include "llvm/Support/raw_ostream.h"
 using namespace llvm;
 
+using namespace SPIRV;
 using namespace spv;
 
 namespace Llpc
 {
+struct ResourceNodeDataSet
+{
+    std::vector<ResourceNodeData> nodes;   // Vector of resource mapping nodes
+    std::map<uint32_t, uint32_t> bindingMap;  // Map from binding to index in nodes vector
+};
+
+// =====================================================================================================================
+// Collect resource nodes from sprirv binary
+void ShaderModuleHelper::CollectResourceNodes(
```

Comment:
please don't collect resource in this way.
We only need fragment shader output info when shader module opt is enabled. we can collect them in some LLVM pass, instead of re-parse the whole SPIRV binary.

Reply:
OK, I will move them into SpirvLowerResourceCollect pass.